player/loadfile.c

```diff
@@ -1583,6 +1583,7 @@ static void append_to_watch_history(struct MPContext *mpctx)
     talloc_steal(ctx, dst.start);
     if (!dst.len) {
         MP_ERR(mpctx, "Failed to serialize history entry\n");
+        fclose(history_file);
```

Comment:
Why not move this under the `done:` block instead of individually fclosing everywhere?

```c
done:
    if (history_file != NULL && fclose(history_file) != 0)
        // ...
```

Reply:
changed